tools/common/Application.cs

```diff
@@ -1495,13 +1495,13 @@ public bool IsAOTCompiled (string assembly)
 
 		public IList<string> GetAotArguments (string filename, Abi abi, string outputDir, string outputFile, string llvmOutputFile, string dataFile)
 		{
-			GetAotArguments (filename, abi, outputDir, outputFile, llvmOutputFile, dataFile, out var processArguments, out var aotArguments);
+			GetAotArguments (filename, abi, outputDir, outputFile, llvmOutputFile, dataFile, null, out var processArguments, out var aotArguments);
 			processArguments.Add (string.Join (",", aotArguments));
 			processArguments.Add (filename);
 			return processArguments;
 		}
 
-		public void GetAotArguments (string filename, Abi abi, string outputDir, string outputFile, string llvmOutputFile, string dataFile, out List<string> processArguments, out List<string> aotArguments, string llvm_path = null)
+		public void GetAotArguments (string filename, Abi abi, string outputDir, string outputFile, string llvmOutputFile, string dataFile, bool? isDedupAssembly, out List<string> processArguments, out List<string> aotArguments, string llvm_path = null)
```

Comment:
I think`isDedupAssembly` can be a nonnullable boolean type and it would simplify comparisons further.

Reply:
`GetAotArguments` is used in legacy build as well, where dedup is not supported. Variable `isDedupAssembly` could have the following values:
 - `NULL` means that dedup is not enabled
 - `FALSE` means that `dedup-skip` flag should be passed for all assemblies
 - `TRUE` means that `dedup-include` flag should be passed for a container assembly

Reply:
Oh right, I forgot about that case. Thanks for the explanation. 
Could you please add this to the description of the PR?